diff --git a/facetapi.admin.js b/facetapi.admin.js
old mode 100644
new mode 100755
index 5f124c3..ea0759f
--- a/facetapi.admin.js
+++ b/facetapi.admin.js
@@ -19,6 +19,17 @@ Drupal.behaviors.facetapi = {
       }
     });
 
+    // Ensures ALL show count checkboxes are updated.
+    // @see http://drupal.org/node/735528
+    $('input[name="show_count"]').change(function() {
+      if ($(this).attr('checked')) {
+        $('input[name="show_count"]').attr('checked', 'checked');
+      }
+      else {
+        $('input[name="show_count"]').removeAttr('checked');
+      }
+    });
+
     // Ensures ALL show expanded checkboxes are updated.
     // @see http://drupal.org/node/735528
     $('input[name="show_expanded"]').change(function() {
diff --git a/facetapi.theme.inc b/facetapi.theme.inc
old mode 100644
new mode 100755
index a8871d6..0906a65
--- a/facetapi.theme.inc
+++ b/facetapi.theme.inc
@@ -56,7 +56,7 @@ function theme_facetapi_link_inactive($variables) {
   $variables['text'] = ($sanitize) ? check_plain($variables['text']) : $variables['text'];
 
   // Adds count to link if one was passed.
-  if (isset($variables['count'])) {
+  if (isset($variables['count']) && $variables['count'] !== FALSE) {
     $variables['text'] .= ' ' . theme('facetapi_count', $variables);
   }
 
diff --git a/plugins/facetapi/widget_links.inc b/plugins/facetapi/widget_links.inc
old mode 100644
new mode 100755
index 0ae49b3..ef4a7ad
--- a/plugins/facetapi/widget_links.inc
+++ b/plugins/facetapi/widget_links.inc
@@ -97,6 +97,10 @@ class FacetapiWidgetLinks extends FacetapiWidget {
     foreach ($build as $value => $item) {
       $row = array('class' => array());
 
+      // If the "show_count" setting is selected the, show the count next to the
+      // facet.
+      $item['#count'] = $settings['show_count'] ? $item['#count'] : FALSE;
+
       // Allow adding classes via altering.
       if (isset($item['#class'])) {
         $attributes['class'] = array_merge($attributes['class'], $item['#class']);
@@ -196,6 +200,18 @@ class FacetapiWidgetLinks extends FacetapiWidget {
       ),
     );
 
+    $form['widget']['widget_settings']['links'][$this->id]['show_count'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Show count'),
+      '#default_value' => !empty($this->settings->settings['show_count']),
+      '#description' => t('Show the facet count next to it.'),
+      '#states' => array(
+        'visible' => array(
+          'select[name="widget"]' => array('value' => $this->id),
+        ),
+      ),
+    );
+
     // @see http://drupal.org/node/735528
     if ($this->facet['hierarchy callback']) {
       $form['widget']['widget_settings']['links'][$this->id]['show_expanded'] = array(
@@ -229,6 +245,7 @@ class FacetapiWidgetLinks extends FacetapiWidget {
     return array(
       'soft_limit' => 20,
       'nofollow' => 1,
+      'show_count' => 1,
       'show_expanded' => 0,
     );
   }
