diff --git a/modules/contrib/facetapi/facetapi.admin.inc b/modules/contrib/facetapi/facetapi.admin.inc
index 9fb7022..02d21d0 100644
--- a/modules/contrib/facetapi/facetapi.admin.inc
+++ b/modules/contrib/facetapi/facetapi.admin.inc
@@ -422,6 +422,28 @@ function facetapi_facet_display_form($form, &$form_state, FacetapiAdapter $adapt
     $plugin = new $class($id, $realm, $adapter->getFacet($facet), $facet_settings);
     $plugin->settingsForm($form, $form_state);
   }
+  $form['widget']['title_override'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Override Facet Title %default_title', array('%default_title' => '(' . t('Default') . ': ' . $facet['label'] . ')')),
+    '#default_value' => $facet_settings->settings['title_override'],
+    '#weight' => -12,
+  );
+  $form['widget']['title'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Facet Title'),
+    '#description' => t('To remove or alter the default title prefix: "Filter by @title:", implement the following hook in your theme: !url.', array(
+      '!url' => l('THEME_facetapi_title()', 'http://drupalcontrib.org/api/drupal/contributions!facetapi!facetapi.theme.inc/function/theme_facetapi_title/7', array(
+        'attributes' => array('target' => '_blank'),
+      ))
+    )),
+    '#default_value' => $facet_settings->settings['title'],
+    '#states' => array(
+      'visible' => array(
+        'input[name="title_override"]' => array('checked' => TRUE),
+      ),
+    ),
+    '#weight' => -11,
+  );
 
   $form['widget']['widget'] = array(
     '#type' => 'select',
diff --git a/modules/contrib/facetapi/plugins/facetapi/adapter.inc b/modules/contrib/facetapi/plugins/facetapi/adapter.inc
index 3fa3ee0..8897b3b 100644
--- a/modules/contrib/facetapi/plugins/facetapi/adapter.inc
+++ b/modules/contrib/facetapi/plugins/facetapi/adapter.inc
@@ -779,6 +779,8 @@ abstract class FacetapiAdapter {
         'sort_weight' => array(),
         'sort_order' => array(),
         'empty_behavior' => 'none',
+        'title_override' => 0,
+        'title' => $facet['label'],
         'facet_more_text' => 'Show more',
         'facet_fewer_text' => 'Show fewer',
       );
diff --git a/modules/contrib/facetapi/plugins/facetapi/widget.inc b/modules/contrib/facetapi/plugins/facetapi/widget.inc
index 57347d7..a261a70 100644
--- a/modules/contrib/facetapi/plugins/facetapi/widget.inc
+++ b/modules/contrib/facetapi/plugins/facetapi/widget.inc
@@ -102,8 +102,8 @@ abstract class FacetapiWidget {
     $this->id = $id;
     $this->realm = $realm;
     $this->settings = $settings;
-    $this->settings->settings += $this->getDefaultSettings();
     $this->facet = $facet;
+    $this->settings->settings += $this->getDefaultSettings();
 
     // Set the default key as the facet's alias.
     $this->key = $facet['field alias'];
@@ -112,7 +112,7 @@ abstract class FacetapiWidget {
   /**
    * Initializes the build, must be invoked prior to executing this widget.
    *
-   * This method is called automatically by FacetapiFacet::build() and shoud
+   * This method is called automatically by FacetapiFacet::build() and should
    * rarely be invoked outside of that context. It is responsible for getting
    * the base render array stored in the FacetapiFacet object and appending it
    * to the container that will be returned, sorting the facets, and adding some
@@ -128,7 +128,7 @@ abstract class FacetapiWidget {
 
     // Initialize the render array.
     $this->build = array(
-      '#title' => $this->facet['label'],
+      '#title' => isset($this->settings->settings['title_override']) && !empty($this->settings->settings['title_override']) ? $this->settings->settings['title'] : $this->facet['label'],
       '#description' => $this->facet['description'],
       '#weight' => $this->facet['weight'],
       '#adapter' => $this->facet->getAdapter(),
